src/pathPlanning/Comb.cpp

```diff
@@ -136,6 +136,48 @@ bool Comb::calc(
         return combing_succeeded;
     }
 
+    // Move start and end point inside the optimal comb boundary
+    // Give more tolerancy when calculating move inside positions, because the target points in this case will be on the borders
+    const coord_t max_move_inside_distance_enlarged = max_moveInside_distance2_ * 1.05;
+    size_t start_inside_poly_optimal = NO_INDEX;
+    const bool start_inside_optimal
+        = moveInside(boundary_inside_optimal_, _start_inside, inside_loc_to_line_optimal_.get(), start_point, start_inside_poly_optimal, max_move_inside_distance_enlarged);
+
+    size_t end_inside_poly_optimal = NO_INDEX;
+    const bool end_inside_optimal
+        = moveInside(boundary_inside_optimal_, _end_inside, inside_loc_to_line_optimal_.get(), end_point, end_inside_poly_optimal, max_move_inside_distance_enlarged);
+
+    size_t start_part_boundary_poly_idx_optimal{};
+    size_t end_part_boundary_poly_idx_optimal{};
+    size_t start_part_idx_optimal
+        = (start_inside_poly_optimal == NO_INDEX) ? NO_INDEX : parts_view_inside_optimal_.getPartContaining(start_inside_poly_optimal, &start_part_boundary_poly_idx_optimal);
+    size_t end_part_idx_optimal
+        = (end_inside_poly_optimal == NO_INDEX) ? NO_INDEX : parts_view_inside_optimal_.getPartContaining(end_inside_poly_optimal, &end_part_boundary_poly_idx_optimal);
+
+    CombPath result_path;
+    bool comb_result;
+
+    if (start_inside_optimal && end_inside_optimal && start_part_idx_optimal == end_part_idx_optimal)
+    {
+        SingleShape part = parts_view_inside_optimal_.assemblePart(start_part_idx_optimal);
+        comb_paths.emplace_back();
+
+        comb_result = LinePolygonsCrossings::comb(
+            part,
+            *inside_loc_to_line_optimal_,
+            start_point,
+            end_point,
+            result_path,
+            -offset_dist_to_get_from_on_the_polygon_to_outside_,
+            max_comb_distance_ignored,
+            fail_on_unavoidable_obstacles);
+        Comb::moveCombPathInside(boundary_inside_minimum_, boundary_inside_optimal_, result_path, comb_paths.back()); // add altered result_path to combPaths.back()
+        // If the endpoint of the travel path changes with combing, then it means that we are moving to an outer wall
+        // and we should unretract before the last travel move when travelling to that outer wall
+        unretract_before_last_travel_move = comb_result && end_point != travel_end_point_before_combing;
+        return comb_result;
+    }
```

Comment:
The block stopping here is sort-of repeated above. (You said there was a third instance as well as I recall.)

Reply:
Yes indeed, but I don't understand 100% of this code so I am a bit afraid of changing the other part to factorize them :grimacing:

Reply:
That's exactly why I'd like to keep it as-is for the patch-release :-)